### PR TITLE
update to react native 76

### PR DESCRIPTION
**Pull request by tyler-whitman**
Update to react native 76 and expo 52

- Android `minSdkVersion` is now 24 (Android 7) up from 21 (Android 5)
- Android `targetSdkVersion` is now 34
- Android `compileSdkVersion` is now 35
- iOS min is now 15.1 
- requires node 20
- requires jdk 17
- requires yarn berry

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -2,22 +2,21 @@
...
     ext {
-        buildToolsVersion = "33.0.0"
-        minSdkVersion = 21
-        compileSdkVersion = 33
-        targetSdkVersion = 33
-
-        // We use NDK 23 which has both M1 support and is the side-by-side NDK version from AGP.
-        ndkVersion = "23.1.7779620"
+        buildToolsVersion = "35.0.0"
+        minSdkVersion = 24
```
Comment (Perronef5) [inline android/build.gradle:6]: Do we care that we might be forcing users to upgrade android version?

File: ios/Helium/Info.plist [hunk 2]
```diff
@@ -72,7 +70,7 @@
 	<string>LaunchScreen</string>
 	<key>UIRequiredDeviceCapabilities</key>
 	<array>
-		<string>armv7</string>
+		<string>arm64</string>
```
Comment (Perronef5) [inline ios/Helium/Info.plist:73]: What does this do?

**Review by Perronef5 — APPROVED** [review]

File: android/build.gradle (hunk 1 above)
Comment (tyler-whitman) [reply android/build.gradle:6]: Its a required change by the RN upgrade

File: ios/Helium/Info.plist (hunk 2 above)
Comment (tyler-whitman) [reply ios/Helium/Info.plist:73]: Unclear haha, just saw it in the changelist

**Pull request merged**